### PR TITLE
Blocks mimes from force opening closed doors

### LINKED ISSUE #7286: Mime Invisible Wall can force open any airlock
### Environment
- Game Version: 4121
- OS Version: MacOS

### Detailed description of bug

Mimes are able to force open any airlock without access simply by using Invisible Wall on top of the airlock and then running into it repeatedly. The door will open but the invisible wall blocks your path. However, as soon as the invisible wall despawns there's enough time to pass through the airlock.

### Steps to reproduce

1.  Be Mime
2. Find Airlock you can't open and face it directly
3. Use Invisible Wall ability
4. Repeatedly move into or click on the airlock
5. Watch it magically open
6. Wait for Invisible Wall to time out
7. Gain Access

### PR DESCRIPTION
**Pull request by MaxIsJoe**
fixes #7286

CL: [Fix] fixed an exploit that allowed mimes to force open closed doors with the invisible wall spell.

### CODE REVIEW

**Comment by Bod9001** [comment]
Wouldn't this cause issues with stuff like firing a fire bolt at a door?

**Comment by MaxIsJoe** [comment]
> Wouldn't this cause issues with stuff like firing a fire bolt at a door?

From what i can tell via the code; no. 
It should only block spells that involve manipulating tiles.

**Comment by Bod9001** [comment]
surely a better check would be checking if the  position is blocked from interaction?

**Comment by corp-0** [comment]
Yeah, I'm not sure if I like this solution, I find it worse than the issue.

**Comment by pyrosshade** [comment]
This solution would likely have to be undone in the future. If not by US, then by those who intend to branch from us and have spells that work specifically on doors (Knock Knock).

**Comment by MaxIsJoe** [comment]
the solution only affects one spell we have right now as far as im aware and like i said on discord this fix is for a non issue and I don't think we should argue about "whats the correct way to do this" for something that is already an intentional design.

**Comment by NoooneyDude** [comment]
This means a mime that attempts to spawn a wall in the doorway they've just passed through (obvious strategy to escape pursuers) will get confused as the spell didn't work for some reason, or worse.

Nope, you need to figure out _why_ the door is opening. Slap a breakpoint in the door code where it opens and read the stack trace.

**Comment by corp-0** [comment]
> this fix is for a non issue

If you think it is a non issue then don't fix it

**Comment by MaxIsJoe** [comment]
> > this fix is for a non issue
> 
> If you think it is a non issue then don't fix it

Listen Gillies, I've been spending a lot of time trying to fix this issue and pulling my hair out over matrix code for the past few weeks and after asking for help multiple times on discord to fix this issue with failed answers I finally found the "best" way to personally handle this based on my capabilities.
I'm fixing this issue because it's marked as an issue and despite me saying it's not one it was still left as one so I thought my opinion was just *one* voice in a sea of crowd. 

Don't tell me to not fix issues otherwise I would have stopped fixing issues completely.

**Comment by MaxIsJoe** [comment]
review pls

**Comment by corp-0** [comment]
I guess not allowing closed doors is good enough **for now**. No idea if we ever want to make a spell that needs a closed door in the future.

**Review by corp-0 — DISMISSED** [review]

File: UnityProject/Assets/Scripts/Systems/Spells/Spell.cs [hunk 1]
```diff
@@ -165,6 +167,11 @@ IEnumerator DespawnAfterDelay()
 					var matrixInfo = MatrixManager.AtPoint(castPosition, true);
 					var localPos = MatrixManager.WorldToLocalInt(castPosition, matrixInfo);
 
+					if (matrixInfo.Matrix.Get<DoorMasterController>(localPos, true).Any(door => door.IsClosed))
```
Comment (Bod9001) [inline UnityProject/Assets/Scripts/Systems/Spells/Spell.cs:170]: just add big old note next to it so if anyone looking for the code sees it instantly

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Systems/Spells/Spell.cs:170]: Forgot to commit it, anyways merge pls.

**Review by corp-0 — APPROVED** [review]

**Review by NoooneyDude — APPROVED** [review]

**Pull request merged**